mirascope/core/base/toolkit.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+from abc import ABC
+from textwrap import dedent
+from typing import Callable, Optional, ClassVar, ParamSpec
+
+from pydantic import BaseModel, ConfigDict
+from typing_extensions import LiteralString
+
+from . import BaseTool
+from .._internal.utils import get_template_variables, convert_function_to_base_tool
+
+_TOOLKIT_TOOL_METHOD_MARKER: LiteralString = "__toolkit_tool_method__"
+
+P = ParamSpec("P")
+
+
+def toolkit_tool(
+        method: Callable[[BaseToolKit, ...], str],
+) -> Callable[[BaseToolKit, ...], str]:
+    # Mark the method as a toolkit tool
+    setattr(method, _TOOLKIT_TOOL_METHOD_MARKER, True)
+
+    return method
+
+
+class BaseToolKit(BaseModel, ABC):
+    """A class for defining tools for LLM call tools."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    _toolkit_tool_method: ClassVar[Callable[[BaseToolKit, ...], str]]
+    _toolkit_template_vars: ClassVar[list[str]]
+
+    def create_tool(self) -> type[BaseTool]:
```

Comment:
should be `create_tools` that returns `list[type[BaseTool]]` for all the tools in the toolkit

Reply:
I understand the intent and will fix it.